docs/architecture.md

```diff
@@ -35,34 +43,93 @@ The architecture diagram at the beginning of this document uses the Fundamental
+
+All content that is delivered through the pipeline and some content that is served straight from the CDN are stored in Franklin Content Hub on S3. For the structure of the content hub, see the [Content Hub](#content-hub) section below.
+### Cloudflare Delivery Stack
+
+The Cloudflare Delivery Stack is the secondary delivery stack for Franklin 4. It is responsible for delivering the content to the end user.
+
+![](./Franklin%20Architecture/Franklin%204%20Stack%20(Cloudflare).png)
 
-Authoring in Franklin 3 is handled through the interplay of three components:
+The biggest architectural difference is that in Cloudflare, the inner CDN and and the pipeline are combined in a single service that is using a Cloudflare Worker.
+#### EDGE on Cloudflare
+
+Like on AWS, the customer can use their own CDN, and opt-in selectively by using `*.hlx-secondary.live` as the origin.
+
+The inner CDN on Cloudflare uses Worker KV instead of edge dictionaries.
+
+Serving media is almost the same as on Fastly, the differences in the implementation of the image optimization features have been abstracted away.
+
+Serving static content is the same as on Fastly.
+
+The code for the outer CDN on Cloudflare is in https://github.com/adobe/helix-cloudflare-live
+
+#### PIPE on Cloudflare
+
+The code for the inner CDN on Cloudflare as well as the worker code for the pipeline on Cloudflare is in https://github.com/adobe/helix-cloudflare-page
+
+All access to the Content Hub is using R2.
+### Content Hub
+
+The Content Hub is a content store optimized for delivery that uses Amazon S3 and Cloudflare R2 (which share an API). It is made up of following constituent parts:
+
+- Media Bus (content addressable storage for images, videos and other media that can be served directly from the CDN with minimal processing)
+  - within the media bus, there is one folder for each `contentBusID`, so that we can delete all content for a given customer when requested
+    - within each `contentBusID` folder, files are addressed by a content hash, so that duplicate content is not stored twice within a given `contentBusID` folder
+- Content Bus (path addressable storage for structured and semistructured authored content)
+  - within the content bus, there is one folder for each `contentBusID`, so that we can delete all content for a given customer when requested
+    - within each `contentBusID` folder, there are two content bus partitions: `preview` and `live`. These content bus partitions separate content that can be served only on `*.hlx.page` from content that is published on `*.hlx.live`. The actual process of publishing involves copying content from `preview` to `live` and purging the attached caches.
+      - each content bus partition has folders that reflect the folder structure of the source repository (Sharepoint or Google Drive)
+        - within each folder, there is a Markdown document for each published document and a JSON file for each published spreadsheet
```

Comment:
- documents can also be marked with redirect information, which causes the pipeline to respond with a redirect instead of the document. redirects objects can be empty or retain the former content.

Reply:
```suggestion
        - within each folder, there is a Markdown document for each published document and a JSON file for each published spreadsheet
        - documents can be marked with redirect metadata, in which case the pipeline will serve a redirect instead of the requested document
```